packages/botonic-react/src/components/carousel.jsx

```diff
@@ -35,14 +57,81 @@ export const Carousel = props => {
     ...{ enable: true, autoHide: true },
     ...getThemeProperty('scrollbar'),
   }
+  const [hasLeftArrow, setLeftArrow] = useState(false)
+  const [hasRightArrow, setRightArrow] = useState(true)
+  const carouselRef = useRef(null)
+  const CustomCarouselLeftArrow = getThemeProperty(
+    'carousel.arrow.left',
+    undefined
+  )
+  const CustomCarouselRightArrow = getThemeProperty(
+    'carousel.arrow.right',
+    undefined
+  )
+  const enableCarouselArrows = getThemeProperty('carousel.enableArrows', true)
+
+  const scrollCarouselBy = value => {
+    carouselRef.current.scrollBy({
+      left: value,
+      behavior: 'smooth',
+    })
+  }
+
+  const setArrowsVisibility = event => {
+    const carousel = event.currentTarget
+    carousel.scrollLeft == 0 ? setLeftArrow(false) : setLeftArrow(true)
+    carousel.scrollLeft == carousel.scrollWidth - carousel.offsetWidth
+      ? setRightArrow(false)
+      : setRightArrow(true)
+  }
+
+  const getArrows = () => {
```

Comment:
By having the c. element width and the margins, then you would be able to do something like:
`const scrollBy =WEBCHAT.DEFAULTS.ELEMENT_WIDTH + WEBCHAT.DEFAULTS.ELEMENT_MARGIN_RIGHT`
and use them in the corresponding `onClick`s like: `onClick={() => scrollCarouselBy(scrollBy)}` or`-scrollBy`